### PR TITLE
[WIP] CI/Tests fixes

### PR DESCRIPTION
**Pull request by Salakar**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,14 @@ coverage
...
+
+# IntelliJ IDEs
+.idea
+
+# VisualStudioCode IDEs
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```
Comment (BridgeAR) [inline .gitignore]: Nit: adding all IDE parts is not possible. Thus, it's normally ideal to let users ignore these globally on their local machine.

File: test/pubsub.spec.js [hunk 2]
```diff
@@ -515,19 +515,15 @@ describe('publish/subscribe', function () {
...
+                                if (typeof pattern === 'string') {
                                     pattern = new Buffer(pattern);
                                     channel = new Buffer(channel);
                                 }
                                 assert.strictEqual(pattern.inspect(), new Buffer('*').inspect());
                                 assert.strictEqual(channel.inspect(), new Buffer('/foo').inspect());
                                 sub.quit(end);
                             });
-                            if (parser === 'javascript') {
-                                assert.notStrictEqual(sub.message_buffers, sub.buffers);
```
Comment (BridgeAR) [inline test/pubsub.spec.js]: Should this assertion not be kept?

File: .gitignore (hunk 1 above)
Comment (Salakar) [reply .gitignore]: It's fairly common practice to add these in on OSS repositories, though I'll get rid of the `!.vscode*` lines as they're not required

File: test/pubsub.spec.js (hunk 2 above)
Comment (Salakar) [reply test/pubsub.spec.js]: Good catch, would have given my code a review before merging don't worry 😅 

I've put this back locally however it causes a failure;

image


This line though doesn't make sense as to what it's testing, do you have any idea what it's trying to test?

Comment (Salakar) [reply test/pubsub.spec.js]: Figured out the issue

**Pull request merged**